pallets/inflation/src/lib.rs

```diff
@@ -322,28 +346,100 @@ pub mod pallet {
+        pub(crate) fn readjusted_config() -> InflationConfiguration {
+            // 1. First calculate the params needed to derive the `max_emission` value used to calculate the current inflation config.
+            let config = ActiveInflationConfig::<T>::get();
+
+            // This is reuse from `new_config` function.
+            let blocks_per_cycle = Balance::from(T::CycleConfiguration::blocks_per_cycle().max(1));
+            let build_and_earn_eras_per_cycle =
+                Balance::from(T::CycleConfiguration::build_and_earn_eras_per_cycle().max(1));
+            let periods_per_cycle =
+                Balance::from(T::CycleConfiguration::periods_per_cycle().max(1));
+
+            // 2. Calculate reward pool amounts per cycle from the existing inflation configuration.
+            let collator_reward_pool = config
+                .collator_reward_per_block
+                .saturating_mul(blocks_per_cycle);
+
+            let treasury_reward_pool = config
+                .treasury_reward_per_block
+                .saturating_mul(blocks_per_cycle);
+
+            let dapp_reward_pool = config
+                .dapp_reward_pool_per_era
+                .saturating_mul(build_and_earn_eras_per_cycle);
+
+            let base_staker_reward_pool = config
+                .base_staker_reward_pool_per_era
+                .saturating_mul(build_and_earn_eras_per_cycle);
+            let adjustable_staker_reward_pool = config
+                .adjustable_staker_reward_pool_per_era
+                .saturating_mul(build_and_earn_eras_per_cycle);
+
+            let bonus_reward_pool = config
+                .bonus_reward_pool_per_period
+                .saturating_mul(periods_per_cycle);
+
+            // 3. Sum up all values to get the old `max_emission` value.
+            let max_emission = collator_reward_pool
+                .saturating_add(treasury_reward_pool)
+                .saturating_add(dapp_reward_pool)
+                .saturating_add(base_staker_reward_pool)
+                .saturating_add(adjustable_staker_reward_pool)
+                .saturating_add(bonus_reward_pool);
+
+            // 4. Calculate new inflation configuration
+            Self::new_config(config.recalculation_era, max_emission)
+        }
+
+        // Calculate new inflation configuration, based on the provided `max_emission`.
+        fn new_config(
+            recalculation_era: EraNumber,
+            max_emission: Balance,
+        ) -> InflationConfiguration {
+            let params = InflationParams::<T>::get();
+
+            // Invalidated parameter, should be cleaned up in the future.
+            // The reason for it's invalidity is because since we've entered the 2nd cycle, it's possible for total
+            // issuance to exceed this cap if unclaimed rewards from previous cycle are claimed.
+            //
```

Comment:
@ermalkaleci I've removed the safety cap check because of this.

Reply:
yes but why don't we increase the cap by 2-300% rather than using arbitrary large value?

Reply:
Because it's a guess with no guarantee it's right.

This was intended basically just for the first cycle. I believe during the code review of the pallet inflation PR we (not sure who) discussed that this is a temporary check, and we should get rid of it once solution is proven to be stable.

I understand where your concern is coming from, but I don't see the necessity for this _incorrect_ check anymore.

If we increase the cap by e.g. 300%, then we end up allowing minting of huge amount of tokens anyways.
If cap is derived from the 7%, increasing it 3x would allow minting of ~2.4B ASTR.

Reply:
Actually no, the cap was used to prevent total issuance exceeding the max cap.

If that was to happen, dApp staking reward payouts would stop working.

Reply:
Ah ok you are right, I see it now.